### PR TITLE
Fix sending synthetic event on detach owner without recomposition

### PR DESCRIPTION
**Pull request by MatkovIvan**
## Proposed Changes

- Set need update pointer position flag on `attach`/`detach` owner to trigger it even without recomposition
- Refactor: remove `PointerPositionUpdater` class. The flag moved to `SyntheticEventSender`
- Remove `EmptyLayout` workaround

## Testing

Test: run existing unit tests

### CODE REVIEW

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/PointerPositionUpdater.skiko.kt [hunk 1]
```diff
@@ -1,54 +0,0 @@
...
- * Updater of the current position of pointer.
- *
- * If something happened with Compose content (it relayouted), we need to send an
- * event to it with the latest pointer position. Otherwise, the content won't be updated
- * by the actual relative position of the pointer.
- *
- * For example, it can be needed when we scroll content without moving the pointer, and we need
- * to highlight the items under the pointer.
- */
-internal class PointerPositionUpdater(
```
Comment (igordmn) [inline compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/PointerPositionUpdater.skiko.kt:31]: The purpose of this class was to decouple `SkiaBasedOwner` from `ComposeScene`:

```
ComposeScene
   owns PointerPositionUpdater
   SkiaBasedOwner
     uses PointerPositionUpdater
```

Otherwise we have a circular dependency, which potentially can complicate code in the future. We already has this complication in the past - owner used `scene` everywhere, and it was hard to follow data flow.

`SkiaBasedOwner` still has a link to it for the `ui-test` module, but this link is hard to break right now.

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt [hunk 2]
```diff
@@ -307,7 +307,6 @@ class ComposeScene internal constructor(
         owner.requestDraw = ::requestDraw
         owner.dispatchSnapshotChanges = snapshotChanges::add
         owner.constraints = constraints
-        invalidateIfNeeded()
```
Comment (igordmn) [inline compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt:308]: `requestUpdatePointer` doesn't imply that it will send invalidate, it is its implementation details.

But we must send invalidate, no matter which implementation of `requestUpdatePointer` will be in the future. So, we need to return this explicit call (and place it after requestUpdatePointer).

Sending multiple `invalidates` during actions is allowed. If there are input lags because of prelimanotory invalidations, they should be fixed via calling `postponeInvalidation` inthe root call stack function.

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt [hunk 3]
```diff
@@ -318,6 +317,7 @@ class ComposeScene internal constructor(
         } else {
             owner.focusOwner.releaseFocus()
         }
+        requestUpdatePointer()
```
Comment (igordmn) [inline compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt:322]: Could you write a test for this?

I checked this snippet manually, it works:
```
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.size
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.ExperimentalComposeUiApi
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.input.pointer.PointerEventType
import androidx.compose.ui.input.pointer.onPointerEvent
import androidx.compose.ui.unit.dp
import androidx.compose.ui.window.Popup
import androidx.compose.ui.window.singleWindowApplication
import kotlinx.coroutines.delay

@OptIn(ExperimentalComposeUiApi::class)
fun main() = singleWindowApplication {
    Box(Modifier.fillMaxSize()) {
        var isPopupShowing by remember { mutableStateOf(false) }
        var isHovered by remember { mutableStateOf(false) }

        LaunchedEffect(Unit) {
            delay(3000)
            isPopupShowing = true
        }

        if (isPopupShowing) {
            Popup {
                Box(Modifier
                    .size(100.dp)
                    .background(if (isHovered) Color.Red else Color.Blue)
                    .onPointerEvent(PointerEventType.Enter) { isHovered = true }
                    .onPointerEvent(PointerEventType.Exit) { isHovered = false }
                )
            }
        }
    }
}
```

But it seems it works even if we remove `requestUpdatePointer`.

Comment (igordmn) [reply compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt:322]: It still doesn't work for hover in `clickable`:
```
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.size
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import androidx.compose.ui.window.Popup
import androidx.compose.ui.window.singleWindowApplication
import kotlinx.coroutines.delay

fun main() = singleWindowApplication {
    Box(Modifier.fillMaxSize()) {
        var isPopupShowing by remember { mutableStateOf(false) }

        LaunchedEffect(Unit) {
            delay(3000)
            isPopupShowing = true
        }

        if (isPopupShowing) {
            Popup {
                Box(Modifier.size(100.dp).background(Color.Red).clickable {  })
            }
        }
    }
}
```

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/PointerPositionUpdater.skiko.kt (hunk 1 above)
Comment (igordmn) [reply compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/PointerPositionUpdater.skiko.kt:31]: As an alternative, we can move `requestUpdatePointer` to `SyntheticEventSender` and pass `SyntheticEventSender` to `SkiaBasedOwner`.

File: compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt (hunk 3 above)
Comment (MatkovIvan) [reply compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt:322]: This PR doesn't change anything with `Popup` because it still uses `EmptyLayout` under the hood

**Review by igordmn — APPROVED** [review]
